### PR TITLE
Add project styling

### PR DESCRIPTION
**Pull request by Nicolaswg**
# Main Point 🥇 

🦅 
> In this new PR we add all the CSS styles for each React component using the module CSS styling feature as well as updating the Redux store with a new data feature.

## Project Schedule 📄 
- Add Project styling.
- Use the information detailed in Zeplin to style the website.
- Follow best CSS practices.

### Section of the Project ✏️ 
<h2>Book Component</h2>

- Use React-toastify feature [Toastify](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwiEicLB_5n2AhV1SzABHXn0BacQFnoECAsQAQ&url=https%3A%2F%2Fwww.npmjs.com%2Fpackage%2Freact-toastify&usg=AOvVaw1bVBaAwuDzB6-e9ANcV1Ve).
- Update form inputs and basic validation.

## Set up 💻 
- Clone the PR repo Branch https://github.com/Nicolaswg/react-bookstore.git
- Open the cloned repo command line and type `npm install`
- Run `npm start` and check your browser at port(3000)
- Optional (Install redux dev tool extension for [Google](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en&authuser=2) &&[ FireFox](https://addons.mozilla.org/en-US/firefox/addon/reduxdevtools/))
- Optional (install an extension for accessing to API request [Google](https://chrome.google.com/webstore/detail/advanced-rest-client/hgmloofddffdnphfgcellkdfbfbjeloo?hl=es-419) && [Firefox](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwih3sGnqJf2AhWhRjABHayuCg8QFnoECBAQAQ&url=https%3A%2F%2Faddons.mozilla.org%2Fen-US%2Ffirefox%2Faddon%2Frestclient%2F&usg=AOvVaw0P4vBiOKQIhK4d-Gc-PQKv) )

<h2> 😄 😁  </h2>

### CODE REVIEW

File: src/components/styles/BookForm.module.css [hunk 1]
```diff
@@ -1,3 +1,58 @@
 * {
```
Comment (Meltrust) [inline src/components/styles/BookForm.module.css:1]: According to the project requirements:

image

Therefore:

- [ ] Please, make the form widths the same width as the book items as in the given design.
- [ ] Kindly add the semi-transparent vertical lines that appear on the given design.
- [ ] Please, match the font styling on the design as much as you can.  

Look:

Your Implementation | Zeplin Design
:---:|:---:
image | image

You can achieve the required styling by going to the Zeplin design, selecting an element, and looking to the right:  almost all the exact CSS code is there:


image

That way, your project will look like the Zeplin design, and you will be able to provide a software CSS solution to a given business requirement. 💪

File: src/components/styles/BookForm.module.css [hunk 2]
```diff
@@ -1,3 +1,58 @@
...
   margin: 0;
 }
+
```
Comment (Meltrust) [inline src/components/styles/BookForm.module.css:4]: - [ ] On your `styles/` folder, you are repeating several CSS rules on the same files.  This happens on all of them `Book.module.css`, `BookForm.module.css`, `BookStore.module.css`, and `Navbar.module.css`. 

Please, don't repeat your CSS rules on each file, to keep your code DRY.

## Explanation

This is an example of the issue on one of the 4 files:

image

## Hints

There are 2 solutions for this issue according to this [support article](https://speckyboy.com/good-bad-css-practices/):

1. Use a class instead:

image

2. Combine the elements:

image


If you implement this, you will achieve the best CSS practices like a pro 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @Nicolaswg,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- Beautiful  project! ✔️ 
- The app works well ✔️ 
 
You can do it

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/styles/BookForm.module.css (hunk 2 above)
Comment (Nicolaswg) [reply src/components/styles/BookForm.module.css:4]: thanks a lot for your review. The best one so far in my microverse journey. Great job

I will make the changes as soon as possible

Comment (Meltrust) [reply src/components/styles/BookForm.module.css:4]: Thanks a lot for your effort, and for your kind words. 🙏

Comment (Nicolaswg) [reply src/components/styles/BookForm.module.css:4]: Hi @Meltrust I made the changes but now a have a weird error that it wasn't in .. 😖 Linter crash with the CSS classes  with a weird array key prop error in some line of code that doesn't have any array ❓

File: src/components/styles/Book.module.css [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+:root {
+  --color-primary: #0290ff;
+  --color-primary-dark: #4386bf;
+  --color-primary-white: #fff;
+  --color-primary-dark-white: #e8e8e8;
+}
+
+.container {
+  display: flex;
```
Comment (moise10r) [inline src/components/styles/Book.module.css:15]: - [ ] [OPTIONAL] This is great using flexbox to layout your content, congrats 👏 , But as said by the previous reviewer, you can still improve this code by using DRY (don't repeat yourself) but why? well, I realized that you have used the property `display: flex` at multiple places, please let's keep the code clean by extracting the property into a separate class that can be reused at multiple places, see the example below: 

- CSS file
```
.d-flex {
  display: flex;
}
```
- HTML file
```
<div class="d-flex">{your content}</div>
```

And remember you can do this for all the other properties that seem to be repeated multiple times, With this, I promise you'll be coding like a pro 😉

**Review by moise10r — APPROVED** [review]
<h1>Status: Approve  :champagne:  :+1: :confetti_ball: 

### **Hi @Nicolaswg** , 

### **Congratulations! 🎉 Project Approved ✅**

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- Your Design looks awesome, keep rocking :rocket: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [ OPTIONAL] Let's improve the styling. The font-family for the category should be `Montserrat`. For the book title and the three buttons below the title, it should be `RobotoSlab` according to the design,

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @moise10r when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/styles/Book.module.css (hunk 3 above)
Comment (Nicolaswg) [reply src/components/styles/Book.module.css:15]: thanks a lot, @moise10r for your review I have a question I can make a single flex property and imported it to all the CSS files?

Comment (moise10r) [reply src/components/styles/Book.module.css:15]: Hi @Nicolaswg,
Yeah, you can do it only if you use CSS `variable `  like this :point_down: 

- CSS
```
:root {
  --primary: #d3d3d3;
  --secondary: #ec5242;
  --text-color: #272a31;
  --background: rgba(230, 230, 230, 0.479);
  --weight: bold;
}



header nav.backg {
  height: 80px;
  background-color: var(--primary);
}

nav .nav_menu li:last-child {
  padding: 4px 14px;
  color: var(--secondary);
  border: 5px solid var(--secondary);
}
```

**Pull request merged**